creator-node/src/premiumContent/premiumContentAccessChecker.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  isCIDForPremiumTrack,
+  isPremiumContentMatch,
+  isRegisteredDiscoveryNode
+} from './helpers'
+import {
+  CheckAccessArgs,
+  CheckAccessResponse,
+  PremiumContentAccessError
+} from './types'
+import { recoverWallet } from '../apiSigning'
+
+export interface AccessChecker {
+  checkPremiumContentAccess(args: CheckAccessArgs): Promise<CheckAccessResponse>
+}
+
+export class PremiumContentAccessChecker implements AccessChecker {
+  async checkPremiumContentAccess({
```

Comment:
may add a comment here to outline the verification steps

Reply:
that would be awesome